### PR TITLE
[Core] Move DevBaseMonitor in SofaValidation

### PR DESCRIPTION
**Pull request by alxbilger**
Moved in https://github.com/sofa-framework/SofaValidation/pull/1

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/framework/Core/src/sofa/core/DevBaseMonitor.h [hunk 1]
```diff
@@ -35,17 +36,18 @@ namespace sofa::core
...
+// {
+// public:
+//     SOFA_CLASS(DevBaseMonitor, objectmodel::BaseObject);
+//     SOFA_BASE_CAST_IMPLEMENTATION(DevBaseMonitor)
+//
+//     /// Destructor
+//     ~DevBaseMonitor() override {}
+//     /// Compute metrics
+//     virtual void eval() = 0;
+// };
```
Comment (bakpaul) [inline Sofa/framework/Core/src/sofa/core/DevBaseMonitor.h]: If you comment everything, then it is more of a deletion than a depreciation.

Comment (alxbilger) [reply Sofa/framework/Core/src/sofa/core/DevBaseMonitor.h]: image

**Review by fredroy — CHANGES_REQUESTED** [review]
I dont want to be "that guy" but
- DevBaseMonitor is required by components in SofaValidation
- SofaValidation is not compiled by the CI as shown by your PR 🤐

I dont know if we really need to compile SofaValidation by the CI (as we could see it is not really used) but I would advocate to toss this interface directly into SofaValidation and be done with it.

**Comment by alxbilger** [comment]
@fredroy interesting. I did search for this class in the SOFA repo and found nothing. I counted on the CI to warn if it was in an external plugin. But the ultimate check from senior SOFA developers was more reliable 😉

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Review by fredroy — APPROVED** [review]

**Pull request merged**